rvgo/fast/radix.go

```diff
@@ -0,0 +1,444 @@
+package fast
+
+import (
+	"math/bits"
+)
+
+// RadixNode is an interface defining the operations for a node in a radix trie.
+type RadixNode interface {
+	// InvalidateNode invalidates the hash cache along the path to the specified address.
+	InvalidateNode(addr uint64)
+	// GenerateProof generates the Merkle proof for the given address.
+	GenerateProof(addr uint64) [][32]byte
```

Comment:
from @protolambda : 
> You could add a slice as argument, to append the result to, and then return the updated slice header. Then you can allocate a slice with the capacity of the expected proof size once, and write to that in each proof level, to prevent allocations.

Reply:
I modified the function so that GenerateProof takes a slice as argument, and each radix node fills in the proof slice at the correct index. The proof slice is only allocated once in the initial generateProof call (`make([][32]byte, 60)`) to reduce the allocations: https://github.com/ethereum-optimism/asterisc/pull/83/commits/c5a040b6b267ed98e4aafd4125292bc6590277aa